.github/workflows/build.yml

```diff
@@ -79,7 +79,7 @@ jobs:
     runs-on: macos-10.15
     strategy:
       matrix:
-        boards: [arm-12, other, sim]
+        boards: [arm-12, xtensa, avr, risc-v, x86, other, sim]
```

Comment:
These changes need to go to the OS & apps repos as well.

Reply:
Yes, once this is merged I will.

Reply:
I'm actually tempted to directly remove everything but sim here (macOS) case.
What do you think?

Reply:
I was thinking we add a new test list only for macOS with just few configs from different archs.
Of course this begs the question of what configs to choose, but as mentioned in the mailing list, those that test the build system should be enough.
Networking configs usually pack a lot into them.  Maybe we can start by those.

Reply:
The sim list builds quite a few configs already, some of which are about networking. Thinking ahead of an eventual addition of functional tests, I think that the sim config will also do some runtime testing. I'm not sure if testing other arch in macOS would really be worth it.

Reply:
i want to keep at least one config for every available toolchain. (compilers and other tools installed by cibuild.sh)

Reply:
Can we run macOS docker to save the installation time?

Reply:
It's a reasonable compromise to create a specific list for macOS:

1. Each arch except sim select one config is enough
2. Since the build tools and some drivers for sim are different between Ubuntu and macOS, we need more config for sim.

Reply:
@yamt since you use macOS daliy, could you create a well-chosen list for macOS?